### DISCUSSION TITLE
Inconsistent property names for ViewConfig

### DISCUSSION THREAD

**Discussion by crambelsoupy** [perspective-dev/perspective#1364] [Q&A]
### Steps to Reproduce:
 1. Create perspective-viewer in angular app
 2. Restore viewer with config as per ViewConfig, with column_pivots and row_pivots properties
 3. call viewer.save to get the config of the viewer

### Expected Result:
The property names used by the perspective-viewer, and provided by the viewer from the save() method should match. 

### Actual Result:
The save() method returns a config with different property names than defined in the typescript exports, more specifically "row-pivots", and "column-pivots".

### Environment:
Version 0.5.0 running in angular app.

### Other context
I have tried rolling back to using the "-" property names but get a warning in console "Deprecated: "row-pivots" config parameter, please use "row_pivots" instead"

**Comment by sc1f** [comment]
Thanks for the report! This is behavior that is intentionally different between `perspective.js` and `perspective-viewer`, although I agree it's a little unclear.

A year ago we decided to normalize the `perspective.js` API to use underscores instead of dashes, so that config objects could be typed out without using quotes in the key names, and the Python binding could use kwargs (as those don't allow dashes). Since users mainly interact with `perspective-viewer`, however, it kept using dashes in order to preserve the viewer's public API.

In this case, you should look at the [PerspectiveViewerOptions](https://github.com/finos/perspective/blob/master/packages/perspective-viewer/index.d.ts#L42) typing in the Typescript definitions for the viewer. The warnings you are seeing are to ensure backwards compatibility (as the invalid keys are converted to valid ones), and if we make a change to the viewer API in the future, there will be more warnings/conversions for backwards compatibility.

Additionally, I tried calling `restore` with `row_pivots` and `column_pivots` and it simply didn't work - the pivots weren't applied. In this case the viewer's behavior is consistent across the viewer and defined correctly in its typescript bindings.
